linkerd/app/inbound/src/policy/http.rs

```diff
@@ -22,24 +24,41 @@ use std::task;
 pub struct NewHttpPolicy<N> {
     metrics: HttpAuthzMetrics,
     inner: N,
+    default_route_meta: Arc<Meta>,
 }
 
 #[derive(Clone, Debug)]
-pub struct HttpPolicy<T, N> {
+pub struct HttpPolicyService<T, N> {
     target: T,
-    client_addr: Remote<ClientAddr>,
-    tls: tls::ConditionalServerTls,
+    meta: ConnectionMeta,
     policy: AllowPolicy,
     metrics: HttpAuthzMetrics,
     inner: N,
+    default_route_meta: Arc<Meta>,
+}
+
+#[derive(Clone, Debug)]
+struct ConnectionMeta {
+    dst: OrigDstAddr,
+    client: Remote<ClientAddr>,
+    tls: tls::ConditionalServerTls,
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("unauthorized request on route kind={:?} name={:?}", .0.kind(), .0.name())]
```

Comment:
tiny nits: this formats the `kind` and `name` strings using `fmt::Debug` rather than `fmt::Display`, so they'll be quoted and have special characters escaped...do we actually want them to be quoted strings? 

i notice that the `ServerUnauthorized` error formats the server meta differently:
https://github.com/linkerd/linkerd2-proxy/blob/15951c535c6d2cc734cc189adff51964749a69cb/linkerd/app/inbound/src/policy.rs#L33

should we match that here?
```suggestion
#[error("unauthorized request on route {}/{}", .0.kind(), .0.name())]
```

Reply:
i left them quoted in case they're empty. In practice, I don't think this error type will actually get rendered, but I think we can expect that both kind and name won't actually be empty so this change is probably fine.